server/src/main/java/io/druid/server/http/CoordinatorResource.java

```diff
@@ -164,4 +193,69 @@ public String apply(DataSegment segment)
         )
     ).build();
   }
+
+  @GET
+  @Path("/historicalstatus")
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response getHistoricalStorageStatus(
+      @QueryParam("simple") String simple
+  )
+  {
+
+    if (simple != null) {
+      List<String> historicalNodes = Lists.newArrayList(
+          Iterables.transform(
+              serverInventoryView.getInventory(),
+              new Function<DruidServer, String>()
+              {
+                @Override
+                public String apply(DruidServer server)
+                {
+                  return server.getHost();
+                }
+              }
+          )
+      );
+
+      return Response.ok(historicalNodes).build();
+    }
+
+    Map<String, Object> statusMap = Maps.newHashMap();
+    for (DruidServer server: serverInventoryView.getInventory()) {
+      try {
+        StatusResponseHolder response = client.go(
```

Comment:
I don't think this is the right approach for this data. A monitoring system should probe the historicals directly, or if the coordinator needs to be that monitoring system for some reason, this needs to be in a special monitoring class.

We should not have what is effectively a proxy hidden within a rest response. And CERTAINLY not have blocking items over a bunch of nodes in a loop, that is ripe for

1. Taking forever when you have hundreds of nodes.
2. Blocking on one mis-behaving node.